app/src/main/kotlin/com/flxrs/dankchat/data/NotificationService.kt

```diff
@@ -254,10 +263,22 @@ class NotificationService : Service(), CoroutineScope {
             ttsMessageQueue -> TextToSpeech.QUEUE_ADD
             else            -> TextToSpeech.QUEUE_FLUSH
         }
-        val ttsMessage = when (name) {
+        var ttsMessage = when (name) {
             previousTTSUser -> message
             else            -> messageFormat.also { previousTTSUser = name }
         }
+        if (removeURL) {
+            val urlPattern = Pattern.compile("((https?|ftp|gopher|telnet|file|Unsure|http):((//)|(\\\\))+[\\w\\d:#@%/;$()~_?\\+-=\\\\\\.&]*)", Pattern.CASE_INSENSITIVE)
```

Comment:
You can skip the `Pattern.compile` part and directly call `toRegex()` on the string with the option `RegexOption.IGNORE_CASE` I think

Reply:
Also, will this regex match links without a protocol?

Reply:
Both issues fixed in [bf179c0](https://github.com/flex3r/DankChat/pull/309/commits/bf179c04098a4ee140a42cf3f75b150d96c67831)

Reply:
It will now filter out URLs with/without any protocol, with/without any prefix, and also URLs with subdomains/#? added.